test/common/http/conn_manager_impl_test.cc

```diff
@@ -2131,6 +2132,210 @@ TEST_F(HttpConnectionManagerImplTest, TestAccessLogWithInvalidRequest) {
   conn_manager_->onData(fake_input, false);
 }
 
+TEST_F(HttpConnectionManagerImplTest, ConnectionTerminatedIfCodecOverridesStreamError) {
+  setup(false, "");
```

Comment:
nit: is it possible to collapse all of these tests into a helper function somehow? I think they are all pretty similar. This file is already super large and confusing and if we can make it less so that would be good. :)